android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java

```diff
@@ -1018,25 +1015,26 @@ public void processProperties(KrollDict d)
 		} else if (d.containsKey(TiC.PROPERTY_BACKGROUND_COLOR) && !nativeViewNull) {
 			bgColor = TiConvert.toColor(d, TiC.PROPERTY_BACKGROUND_COLOR);
 
-			if (canApplyTouchFeedback(d)) {
-				applyTouchFeedback(bgColor, d.containsKey(TiC.PROPERTY_TOUCH_FEEDBACK_COLOR)
-												? TiConvert.toColor(d, TiC.PROPERTY_TOUCH_FEEDBACK_COLOR)
-												: null);
-			} else {
-				// Set the background color on the view directly only
-				// if there is no border. If a border is present we must
-				// use the TiBackgroundDrawable.
-				if (hasBorder(d)) {
-					if (background == null) {
-						applyCustomBackground(false);
-					}
-					background.setBackgroundColor(bgColor);
-				} else {
-					nativeView.setBackgroundColor(bgColor);
+			// Set the background color on the view directly only
+			// if there is no border. If a border is present we must
+			// use the TiBackgroundDrawable.
+			if (hasBorder(d)) {
+				if (background == null) {
+					applyCustomBackground(false);
 				}
+				background.setBackgroundColor(bgColor);
+			} else {
+				nativeView.setBackgroundColor(bgColor);
 			}
 		}
 
+		if (canApplyTouchFeedback(d)) {
+			applyTouchFeedback(bgColor, background,
+							   d.containsKey(TiC.PROPERTY_TOUCH_FEEDBACK_COLOR)
+								   ? TiConvert.toColor(d, TiC.PROPERTY_TOUCH_FEEDBACK_COLOR)
```

Comment:
I understand that this is way outside the scope of this PR, but this screams to me that we should have a better API to handling cases like this. Specially why not just have `TiConvert.toColor()` handle the case where ether key doesn't exist and have it return a default value (in this case `null`)?

Presumably it's because it returns an `int`? I think it may useful to have a `TiColor` class that can encapsulate the concept of colors and incorporate the `TiColorHelper` logic and support outputting any representations (like an int) that we'd need. In fact, Android itself appears to already have the same concept in `android.graphics.Color`. Bonus is that you can then return `null` in a case like this.

Reply:
We have a few more places where we fallback to a default color, so it would definitely be an improvement to deal with that in one single method. I can try to do it in this PR, but I think a new one dedicated to the refactoring would be better.